clu/src/lib.rs

```diff
@@ -130,6 +133,45 @@ fn linux_init(config: Arc<RwLock<settings::RitaSettingsStruct>>) -> Result<(), E
         },
     }
 
+    match device_option {
+        Some(existing_device) => {
+            info!("Device name is {}", existing_device);
+        }
+        None => {
+            let release_file_path = "/etc/althea-firmware-release";
+            info!(
+                "No device name was found, reading from {}",
+                release_file_path
+            );
+
+            let mut contents = String::new();
+            match File::open(release_file_path) {
+                Ok(mut f) => {
+                    f.read_to_string(&mut contents)?;
+                }
+                Err(e) => warn!("Couldn't open {}: {}", release_file_path, e),
+            };
+
+            for line in contents.lines() {
+                if line.starts_with("device:") {
+                    let mut array = line.split(" ");
```

Comment:
@asoltys (You'll want to ignore this and see the later nit instead, but it's good practice anyway) It's not an array, but an `Iterator`, this could get very confusing!